### ISSUE TITLE
Allow setting the path for the cookie signer.

### ISSUE THREAD

**Issue by HEPOSHEIKKI** [michaelvanstraten/actix-jwt-auth-middleware#24]
Currently, there is no explicit way of setting the designated path for the returned cookie pair, resulting in the cookie not being recognized by browsers.

# Reproduction steps:

Have the following API layout, using the simple example:
    `/v1/auth/login`
    `/v1/api/hello`

Define the App with
```rs
App::new()
    .service(web::scope("/v1")
    .service(login)
    .use_jwt(authority.clone(), web::scope("").service(hello))
```

Retrieve the login cookie from `/auth/login`
Try to access `/v1/api/hello`

As you can see, `hello` will report unauthorized, as the cookie path has been set to `/v1/auth`, which doesn't cover the `api` route.
![image](https://github.com/michaelvanstraten/actix-jwt-auth-middleware/assets/51783206/782af7fa-9f9c-485a-859b-96868343f980)

**Comment by michaelvanstraten** [comment]
Yes, I can see that this is an issue with the [`TokenSigner`](https://docs.rs/actix-jwt-auth-middleware/latest/actix_jwt_auth_middleware/struct.TokenSigner.html#) struct. 

Nonetheless, you still can create your own cookie using the value provided from [`TokenSigner::create_signed_token`](https://docs.rs/actix-jwt-auth-middleware/latest/actix_jwt_auth_middleware/struct.TokenSigner.html#method.create_signed_token).

Something like this should work for you:
```rust
let token = token_signer.create_signed_token(claims, token_lifetime)?;
let cookie = Cookie::build("access_token".to_string(), token)
    .path("/")
    .secure(true)
    .finish();
```

The [`TokenSigner::create_access_cookie`](https://docs.rs/actix-jwt-auth-middleware/latest/actix_jwt_auth_middleware/struct.TokenSigner.html#method.create_access_cookie) method is just a convenience [wrapper](https://docs.rs/actix-jwt-auth-middleware/latest/src/actix_jwt_auth_middleware/token_signer.rs.html#251) around the above that ensures that the `name` of the cookie is set correctly. 

Maybe it would be better to just return a [builder](https://docs.rs/actix-web/latest/actix_web/cookie/struct.CookieBuilder.html) that allows you to set further options on the cookie.

**Comment by szsdk** [comment]
I found that your workaround only partially addresses my issue. My goal is to enable users to log in and gain full access to the /app/ directory. I want to allow cookies to refresh properly. When a user refreshes the page at /app/foo, the new access cookie is set with the path /app/foo, which causes problems as it cannot be overwritten through a new login process. I have no idea about to modify the cookie gotten from `refresh_authorizer`.

**Comment by ovalek** [comment]
I had the same problem with access_token refreshing. I propose to add a method to TokenSigner builder, that would accept closure with CookieBuilder parameter. PR https://github.com/michaelvanstraten/actix-jwt-auth-middleware/pull/26 This would allow us to set Path and other settings.

**Comment by michaelvanstraten** [comment]
[I suggested](https://github.com/michaelvanstraten/actix-jwt-auth-middleware/pull/26#issuecomment-2195246263) to @ovalek that rather than adding a new method that would accept a closure to modify the [`CookieBuilder`](https://docs.rs/cookie/latest/cookie/struct.CookieBuilder.html) before it gets built, we should add an initial `CookieBuilder` as a user-controllable parameter.

Please speak out if you have a use case where a function is required; it could be that I don't see the full picture here.
